### DISCUSSION TITLE
Allow --single-package to be configured in turbo.json

### DISCUSSION THREAD

**Discussion by MisterJimson** [vercel/turborepo#7828] [Help] [answered]
### Goals

I have a monorepo that is setup to be use in --single-package mode. Because I use workspaces for non-turborepo related tasks, I need to use --single-package for every command to avoid turborepo running tasks outside of the root.

I would like to be able to set --single-package mode in turbo.json so it's not required in every CLI command.

### Background

The only the option is to use --single-package on every single turbo command for this project, which is error prone and verbose.

### Proposal

`"mode": "single-project",` could be added as an optional property in turbo.json

**Comment by anthonyshew** [comment]
Can you provide a reproduction repo for what you're describing? I'm a bit confused.

If you're talking about a Single Package Workspace (for example, the result if npx create-next-app or npm create vite), `turbo` will automatically run in Single Package Mode.

**Reply by MisterJimson** [reply]
Ill try to get a reprod.

I have a monorepo where some commands need to be ran like a typical monorepo (execute task in each inner folder/package), and other commands need to be ran at the root, treating the entire monorepo as a single package.

I'd like to be able to set `"mode": "single-project"`, per pipeline ideally, and have the turbo cli set `--single-package` automatically for those commands.

**Reply by anthonyshew** [reply, marked as answer]
Are you describing a Root Task? https://turbo.build/repo/docs/core-concepts/monorepos/running-tasks#running-tasks-from-the-root

**Reply by MisterJimson** [reply]
I missed root tasks when investigating this and it does seem like it's made for this purpose. Thanks.

I notice that root tasks are not documented [here](https://turbo.build/repo/docs/reference/configuration) where I spent most of my time investigating this type of functionality.

**Reply by anthonyshew** [reply]
Thanks for the feedback. I'm working on brand new documentation so I'll make sure we surface this information better in the future. 👍
